server/master/src/main/java/org/apache/accumulo/master/Master.java

```diff
@@ -1202,7 +1205,9 @@ public void run() {
               try {
                 TServerConnection connection = tserverSet.getConnection(server);
                 if (connection != null) {
-                  connection.halt(masterLock);
+                  if (rateLimiter.tryAcquire()) {
```

Comment:
If its not acquired, will the server still be removed from badServers?  I am not quite sure trying to match the curly braces looking at the diff.

Reply:
@keith-turner  Yes, That isn't a change to what happens now. If it fails to issue the shutdown, the sever is removed and status is attempted three more times

Reply:
> That isn't a change to what happens now. If it fails to issue the shutdown, the sever is removed and status is attempted three more times

Oh I see.  That is kinda cool, I really like that behavior.  I think a comment stating that this behavior is intentional would be good.  That if the rate limit does not allow a halt, then we want to reset the counter.